backend/src/main/java/com/mcmasterbaja/analyzer/AnalyzerFactory.java

```diff
@@ -1,88 +1,79 @@
-          return null;
-        }
-        int start = Integer.parseInt((String) options[0]);
-        int end = Integer.parseInt((String) options[1]);
-        return new SplitAnalyzer(inputFiles, inputColumns, outputFiles, start, end);
+  @Inject
+  @AnalyzerType(AnalyzerEnum.LINEAR_MULTIPLY)
+  Analyzer linearMultiplyAnalyzer;
 
-      case LINEAR_MULTIPLY:
-        if (options[1] == "" || options[0] == "") {
-          return null;
-        }
-        double m = Double.parseDouble((String) options[0]);
-        double b = Double.parseDouble((String) options[1]);
-        return new LinearMultiplyAnalyzer(inputFiles, inputColumns, outputFiles, m, b);
+  @Inject
+  @AnalyzerType(AnalyzerEnum.RDP_COMPRESSION)
+  Analyzer rdpCompressionAnalyzer;
 
-      case CONSTANT_ADDER:
-        if (options[3] == "" || options[2] == "" || options[1] == "" || options[0] == "") {
-          return null;
-        }
-        double a1 = Double.parseDouble((String) options[0]);
-        double b2 = Double.parseDouble((String) options[1]);
-        double c1 = Double.parseDouble((String) options[2]);
-        double d1 = Double.parseDouble((String) options[3]);
-        return new ConstantAdderAnalyzer(inputFiles, inputColumns, outputFiles, a1, b2, c1, d1);
+  @Inject
+  @AnalyzerType(AnalyzerEnum.ROLL_AVG)
+  Analyzer rollAvgAnalyzer;
 
-      case AVERAGE:
-        int[] range = new int[2];
-        range[0] = Integer.parseInt((String) options[0]);
-        range[1] = Integer.parseInt((String) options[1]);
-        return new AverageAnalyzer(inputFiles, outputFiles, range);
+  @Inject
+  @AnalyzerType(AnalyzerEnum.SGOLAY)
+  Analyzer sgolayAnalyzer;
 
-      case INTERPOLATER_PRO:
-        return new InterpolaterProAnalyzer(inputFiles, inputColumns, outputFiles);
+  @Inject
+  @AnalyzerType(AnalyzerEnum.SPLIT)
+  Analyzer splitAnalyzer;
 
+  public Analyzer createAnalyzer(AnalyzerEnum type) {
+    switch (type) {
+      case ACCEL_CURVE:
+        return accelCurveAnalyzer;
+      case AVERAGE:
+        return averageAnalyzer;
+      case CONSTANT_ADDER:
+        return constantAdderAnalyzer;
       case CUBIC:
-        double a = Double.parseDouble((String) options[0]);
-        double b1 = Double.parseDouble((String) options[1]);
-        double c = Double.parseDouble((String) options[2]);
-        double d = Double.parseDouble((String) options[3]);
-        return new CubicAnalyzer(inputFiles, inputColumns, outputFiles, a, b1, c, d);
+        return cubicAnalyzer;
+      case INTERPOLATER_PRO:
+        return interpolaterProAnalyzer;
+      case LINEAR_MULTIPLY:
+        return linearMultiplyAnalyzer;
+      case RDP_COMPRESSION:
+        return rdpCompressionAnalyzer;
+      case ROLL_AVG:
+        return rollAvgAnalyzer;
+      case SGOLAY:
+        return sgolayAnalyzer;
+      case SPLIT:
+        return splitAnalyzer;
       default:
-        return null;
+        throw new IllegalArgumentException("Invalid analyzer type");
```

Comment:
By extension, I also find this a bit a confusing since there is now and AnalyzerEnum and AnalyzerType